pilot/pkg/model/config.go

```diff
@@ -260,6 +260,14 @@ type ConfigStoreCache interface {
 
 	// HasSynced returns true after initial cache synchronization is complete
 	HasSynced() bool
+
+	// Share allows one config store to share events with another config store
+	// or as a general mechanism for a config store to notify other config stores of
+	// an event of particular type. This is currently being used by the Kube service
+	// registry to notify the service entry config controller about pod events (in the
+	// form of workload entry configs) so that the service entry controller can process
+	// these for ServiceEntries that select both pods and VMs.
+	Share(kind resource.GroupVersionKind, config *Config, event Event)
```

Comment:
Confused - we already have 3 interface for distributing events (that I know about), why can't it use the existing RegisterEventHandler - i.e. bootstrap registers an event handler on K8S registry, with the handler in service entry ? 

I don't think yet another interface is needed. 

We also have the event distribution in ADS, the different interface for registring instances, and I'm sure more.

Reply:
RegisterEventHandler is not for service registry. It is only for the configstore cache.

Reply:
AppendInstanceHandler is currently used for service registry.

Reply:
And XdsUpdater interface as well - my point was we have 3 inconsistent mechanisms for events, no need for 4.

Reply:
Besides - this seems in the wrong place, why would a store implement a 'Share' - the destination store is not sharing anything. Maybe have it as a separate interface. Usually calling store.Share() means the store is supposed to share something with someone.

Reply:
AppendInstanceHandler is now a stub. It is used only for consul. not for anything else. The xDSUpdater is only receiving EDS updates for a given service. So from inside Kube, there is no way to know which service entry corresponds to a pod. For this, we have to go to the service entry code. The nicer and cleaner way was simply adding a pod notification callback to service entry which you were opposed to. The more elaborate way is what I have done via the shared interface to avoid circular dependencies. But this too seems to be contentious. So I am lost as to what exact interface you want me to use so that k8s can notify service entry code that a pod update has occurred and that any service entry inthe namespace with label selectors needs to process this event.